coresimd/src/x86/i586/sse2.rs

```diff
@@ -582,149 +587,150 @@ pub unsafe fn _mm_or_si128(a: __m128i, b: __m128i) -> __m128i {
 #[target_feature = "+sse2"]
 #[cfg_attr(test, assert_instr(xorps))]
 pub unsafe fn _mm_xor_si128(a: __m128i, b: __m128i) -> __m128i {
-    __m128i::from(i8x16::from(a) ^ i8x16::from(b))
+    simd_xor(a, b)
 }
 
 /// Compare packed 8-bit integers in `a` and `b` for equality.
 #[inline(always)]
 #[target_feature = "+sse2"]
 #[cfg_attr(test, assert_instr(pcmpeqb))]
-pub unsafe fn _mm_cmpeq_epi8(a: i8x16, b: i8x16) -> i8x16 {
-    a.eq(b)
+pub unsafe fn _mm_cmpeq_epi8(a: __m128i, b: __m128i) -> __m128i {
+    mem::transmute::<i8x16, _>(simd_eq(a.as_i8x16(), b.as_i8x16()))
```

Comment:
Is the `::< >` necessary here? All types are fixed, same below.

Reply:
Unfortunately yeah, the [`simd_eq` intrinsic](https://github.com/rust-lang-nursery/stdsimd/blob/6ef0ae27aaa087b97b9482f22eaf8c9ed6cf9853/coresimd/src/simd_llvm.rs#L6) has a generic return type, but I believe that generic return type needs to at least have the same number of lanes as the comparison. That means that we just have to constrain `U` there to say what it needs to be.